lib/request.ts

```diff
@@ -148,90 +130,75 @@ class Request {
   }
 
   /**
-   * Request internal identifier getter
-   * @return {string}
+   * Request internal ID
    */
-  get internalId () {
+  get internalId (): string {
     return this[_internalId];
   }
 
   /**
-   * Request timestamp getter
-   * @returns {number}
+   * Request timestamp (in micro-time)
    */
-  get timestamp () {
+  get timestamp (): number {
     return this[_timestamp];
   }
 
   /**
-   * Request status getter
-   * @returns {number}
+   * Request HTTP status
    */
-  get status () {
+  get status (): number {
     return this[_status];
   }
 
-  /**
-   * Request status setter
-   * @param {number} i - new request status
-   */
-  set status (i) {
+  set status (i: number) {
     this[_status] = assert.assertInteger('status', i);
   }
 
   /**
-   * Request input getter
-   * @returns {RequestInput}
+   * Request input
    */
-  get input () {
+  get input (): RequestInput {
     return this[_input];
   }
 
   /**
-   * Request context getter
-   * @returns {RequestContext}
+   * Request context
    */
-  get context () {
+  get context (): RequestContext {
     return this[_context];
   }
 
   /**
-   * Request error getter
-   * @returns {null|KuzzleError}
+   * Request error
    */
-  get error () {
+  get error (): KuzzleError | null {
     return this[_error];
   }
 
   /**
-   * Request result getter
-   * @returns {null|*}
+   * Request result
    */
-  get result () {
+  get result (): any | null {
     return this[_result];
   }
 
   /**
-   * Request response getter
-   * @returns {{status: (number|*), error: (null|*), requestId: string, controller: string, action: string, collection: string, index: string, volatile: Object, headers: (Array|*), result: (null|*|Object)}}
+   * Request response
    */
-  get response () {
+  get response (): RequestResponse {
     if (this[_response] === null) {
       this[_response] = new RequestResponse(this);
     }
+
     return this[_response];
   }
 
   /**
    * Sets the request status to the error one, and fills the error member
-   *
-   * @name setError
-   * @param {Object} error
-   * @memberOf Request
    */
-  setError(error) {
+  setError (error: KuzzleError) {
     if (!error || !(error instanceof Error)) {
-      throw new InternalError('cannot set non-error object as a request\'s error');
+      throw new InternalError('Cannot set non-error object as a request\'s error');
     }
```

Comment:
This method handles KuzzleError (which extends Error) and native Error objects.
Also, if ts enforces parameter types, do we still need this check?

```suggestion
  setError (error: Error) {
```